src/results.rs

```diff
@@ -0,0 +1,278 @@
+use crate::value::{PyDeserializeValue, PyDeserializedValue};
+use bytes::Bytes;
+use pyo3::exceptions::{PyRuntimeError, PyTypeError};
+use pyo3::prelude::{PyDictMethods, PyModule, PyModuleMethods};
+use pyo3::types::{PyDict, PyString};
+use pyo3::{
+    Bound, IntoPyObject, Py, PyAny, PyErr, PyRefMut, PyResult, Python, pyclass, pymethods, pymodule,
+};
+use scylla::response::query_result::QueryRowsResult;
+use scylla_cql::_macro_internal::{ColumnIterator, FrameSlice};
+use scylla_cql::deserialize::result::RawRowIterator;
+use scylla_cql::frame::response::result::{DeserializedMetadataAndRawRows, ResultMetadataHolder};
+use scylla_cql::value::Row;
+use stable_deref_trait::StableDeref;
+use std::fmt::Write;
+use std::ops::Deref;
+use yoke::{Yoke, Yokeable};
+
+// Originally written by @Lorak-mmk, moved by @patrycja-ziemkiewicz
+#[pyclass]
+pub(crate) struct RequestResult {
+    pub(crate) inner: scylla::response::query_result::QueryResult,
+}
+
+#[pymethods]
+impl RequestResult {
+    fn __str__<'gil>(&mut self, py: Python<'gil>) -> PyResult<Bound<'gil, PyString>> {
+        let mut result = String::new();
+        let rows_result = match self.inner.clone().into_rows_result() {
+            Ok(r) => r,
+            Err(e) => return Ok(PyString::new(py, &format!("non-rows result: {}", e))),
+        };
+        for r in rows_result.rows::<Row>().map_err(|e| {
+            PyRuntimeError::new_err(format!("Failed to deserialize metadata: {}", e))
+        })? {
+            let row = match r {
+                Ok(r) => r,
+                Err(e) => {
+                    return Err(PyRuntimeError::new_err(format!(
+                        "Failed to deserialize row: {}",
+                        e
+                    )));
+                }
+            };
+            write!(result, "|").unwrap();
+            for col in row.columns {
+                match col {
+                    Some(c) => write!(result, "{}", c).unwrap(),
+                    None => write!(result, "null").unwrap(),
+                };
+                write!(result, "|").unwrap();
+            }
+            writeln!(result).unwrap();
+        }
+        Ok(PyString::new(py, &result))
+    }
```

Comment:
:open_mouth:  Wow, is this essentially a query result displayer? Do we want such thing? @Lorak-mmk